front-packages/akeneo-design-system/src/components/Collapse/Collapse.tsx

```diff
@@ -73,13 +76,23 @@ const Collapse = React.forwardRef<HTMLDivElement, CollapseProps>(
     forwardedRef: Ref<HTMLDivElement>
   ) => {
     const [contentHeight, setContentHeight] = useState<number>(0);
+    const [overflow, setOverflow] = useState<string>(isOpen ? 'inherit' : 'hidden');
     const contentRef = useRef<HTMLDivElement>(null);
+    const isMounted = useIsMounted();
 
     const handleCollapse = () => onCollapse(!isOpen);
 
     useEffect(() => {
-      setContentHeight(isOpen && null !== contentRef.current ? contentRef.current.scrollHeight : 0);
-    }, [isOpen]);
+      if (contentRef.current && isMounted()) {
```

Comment:
```suggestion
      if (contentRef.current) {
```

For me we don't need to check isMounted here => if useEffect is called the component is necessarily mounted. We only need it in asynchronous process no ?

Reply:
Indeed